shared/src/main/scala/mlscript/ucs/stages/Desugaring.scala

```diff
@@ -285,20 +292,30 @@ trait Desugaring { self: PreTyper =>
     def rec(scrutineeVar: Var, split: s.PatternSplit)(implicit scope: Scope): c.Split = split match {
       case s.Split.Cons(head, tail) => 
         head.pattern match {
-          case s.AliasPattern(nme, pattern) => ???
+          case pattern @ s.AliasPattern(_, _) =>
+            raiseError(PreTyping, msg"alias pattern is not supported for now" -> pattern.toLoc)
+            rec(scrutineeVar, tail)
           case s.LiteralPattern(literal) =>
-            val test = context.freshTest().withFreshSymbol
-            c.Split.Let(
-              rec = false,
-              name = test,
-              term = mkBinOp(scrutineeVar, Var("==="), literal, true),
-              tail = c.Branch(
-                scrutinee = test,
-                pattern = truePattern,
-                continuation = desugarTermSplit(head.continuation)(PartialTerm.Empty, scope + test.symbol, context)
-              ) :: rec(scrutineeVar, tail)
-            )
-          case s.ConcretePattern(nme) => 
+            scrutineeVar.getOrCreateScrutinee
+                        .withAlias(scrutineeVar)
```

Comment:
No horizontal alignment plz. We should be able to rename variables without having to change other code's indentation.

Reply:
Done.